### PR TITLE
update dockerfile for ubuntu22.04.

### PR DESCRIPTION
**Pull request by ForLisiteng**

### CODE REVIEW

File: docker/Dockerfile.ubuntu22 [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    reprepro \
+    unzip \
+    libjsoncpp-dev \
+    uuid-dev\
+    liblog4cplus-dev\
+    dnsutils
+
+# Install the SDK
+WORKDIR /opt/intel
+RUN wget https://download.01.org/intel-sgx/sgx-dcap/1.19/linux/distro/ubuntu20.04-server/sgx_linux_x64_sdk_2.22.100.3.bin \
```
Comment (yang8621) [inline docker/Dockerfile.ubuntu22:50]: ubuntu20.04-server ?      there should be a 22.04 version

File: docker/Dockerfile.ubuntu22 [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    && chmod 777 sgx_linux_x64_sdk_2.22.100.3.bin \
+    && sh -c 'echo yes | ./sgx_linux_x64_sdk_2.22.100.3.bin'
+
+#Install SDK toolset
+RUN wget https://download.01.org/intel-sgx/sgx-linux/2.22/as.ld.objdump.r4.tar.gz \
+    && tar -zxf as.ld.objdump.r4.tar.gz \
+    && cp external/toolset/ubuntu20.04/* /usr/local/bin
```
Comment (yang8621) [inline docker/Dockerfile.ubuntu22:57]: and here

File: docker/Dockerfile.ubuntu22 [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+# Install DCAP packages
+# DCAP repository setup
+RUN wget https://download.01.org/intel-sgx/sgx-dcap/1.19/linux/distro/ubuntu20.04-server/sgx_debian_local_repo.tgz \
```
Comment (yang8621) [inline docker/Dockerfile.ubuntu22:61]: and here

File: docker/Dockerfile.ubuntu22 (hunk 1 above)
Comment (yang8621) [reply docker/Dockerfile.ubuntu22:50]: and should be version 1.20?

File: docker/Dockerfile.ubuntu22 [hunk 4]
```diff
@@ -0,0 +1,87 @@
+FROM ubuntu:22.04
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+ARG TAG_VERSION=main
+
+RUN apt-get update && apt-get install -y \
```
Comment (yang8621) [inline docker/Dockerfile.ubuntu22:7]: liblog4cplus-1.1-9 and   libssl1.1 not required?

File: docker/Dockerfile.ubuntu22 [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    dnsutils
+
+# Install the SDK
+WORKDIR /opt/intel
+RUN wget https://download.01.org/intel-sgx/sgx-dcap/1.19/linux/distro/ubuntu20.04-server/sgx_linux_x64_sdk_2.22.100.3.bin \
+    && chmod 777 sgx_linux_x64_sdk_2.22.100.3.bin \
+    && sh -c 'echo yes | ./sgx_linux_x64_sdk_2.22.100.3.bin'
+
+#Install SDK toolset
+RUN wget https://download.01.org/intel-sgx/sgx-linux/2.22/as.ld.objdump.r4.tar.gz \
```
Comment (yang8621) [inline docker/Dockerfile.ubuntu22:55]: 2.23

File: docker/Dockerfile.ubuntu22 (hunk 4 above)
Comment (ForLisiteng) [reply docker/Dockerfile.ubuntu22:7]: liblog4cplus-1.1-9 - > liblog4cplus-dev, libssl1.1 -> libssl-dev

**Review by yang8621 — APPROVED** [review]

**Review by yang8621 — APPROVED** [review]

**Review by syan10 — APPROVED** [review]

**Pull request merged**